packages/data/src/registry.js

```diff
@@ -17,11 +17,21 @@ import createCoreDataStore from './store';
  *
  * @typedef {WPDataRegistry}
  *
- * @property {Function} registerGenericStore
- * @property {Function} registerStore
- * @property {Function} subscribe
- * @property {Function} select
- * @property {Function} dispatch
+ * @property {Function} registerGenericStore Given a namespace key and settings
+ *                                           object, registers a new generic
+ *                                           store.
+ * @property {Function} registerStore        Given a namespace key and settings
+ *                                           object, registers a new namespace
+ *                                           store.
+ * @property {Function} subscribe            Given a function callback, invokes
```

Comment:
Using this property as an example because it's the easiest one to do. IMHO, it would be better to do one of the following options since the type `Function` is not really helpful or descriptive.

### Option 1

```js
/**
 * @callback Subscriber Description of subscriber
 * @param {() => void} callback Description of callback parameter.
 * @return {void}
 */
```

### Option 2

```js
/**
 * @typedef {(callback: () => void)} Subscriber Description of subscriber
 */
```

And then use it on this line as...

```js
/**
 * [...]
 * @prop {Subscriber} registerGenericStore Given a namespace key and settings
 *                                         object, registers a new generic store.
 * [...]
 */
```

------

I assume that this is also temporary though until typescript checking makes its way here. So not _super_ important.

Reply:
Are those syntaxes standard in JSDoc? It would be a good thing to detail in the [documentation standards](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/), I think.

In general, I agree with you. I didn't aim to change the types here, however.

Reply:
Yes.

I agree that the documentation standards should be combed through for accuracy and, ideally, these standards should be linted automatically. Pretty sure `eslint-plugin-jsdoc` should be able to do the brunt of this.